subsys/net/lib/mqtt_helper/Kconfig

```diff
@@ -12,6 +12,12 @@ menuconfig MQTT_HELPER
 
 if MQTT_HELPER
 
+config MQTT_HELPER_IPV4_ONLY
```

Comment:
I'm not really sure if we need such an option.

> may be useful if a broken IPv6 implementation is encountered, e.g. resolves IPv6 addresses but doesn't have routing.

Is this really something one can encounter in the wild?

Reply:
I wanted to be backwards compatible because this is a change to an internal library. I don't really know why it was written as IPv4 only in the first place.

I don't like the option, so happy to remove it.

There is another workaround that could be used, at least for the Thingy:91 you can configure the packet data network to be IPv4 only rather than dual stack (an existing configuration option). There is probably something similar for WiFi/other devices to remove IPv6 if needed.

Reply:
FYI, even tough the PDP context is configured for a specific IP family on the device side. The network may override this.
cc @stig-bjorlykke

So in reality, you don't know what you get. Just that its more likely that you get what you ask for, I guess.